assembly/ScaleString.ts

```diff
@@ -1,41 +1,44 @@
 import { Bytes } from "./utils/Bytes"
+import { ByteArray } from "./ByteArray";
+
+export class ScaleString extends ByteArray {
+
+    private readonly value: string;
 
-export class ScaleString extends Array<u8> {
     constructor(input: string) {
+        super([]);
+        this.value = input;
+
         const inputBuffer: ArrayBuffer = String.UTF8.encode(input);
         const u8Input = Uint8Array.wrap(inputBuffer)
```

Comment:
Why dont you pass the `u8Input` to the `super` and let the `BytesArray` populate the values?